shared/repository/lib/api/documents/ContactRepository.mjs

```diff
@@ -0,0 +1,150 @@
+import RepoBase from '../RepoBase.mjs';
+import { RepoError } from '../models/index.mjs';
+
+export class ContactRepository extends RepoBase {
+    /**
+     * @typedef {Class} ContactRepository
+     */
+
+    /**
+     * @param {Object} options - repository options
+     * @param {Object} options.db - databse connection
+     */
+    constructor(options) {
+        const { db } = options;
+
+        super(db);
+    }
+
+    /**
+     * @method
+     * @param {number} userId - user identifier
+     * @param {string} trackedPhone - contact added by user that should be tracked
+     * @return {Promise<Object>} returns data saved in DB
+     */
+    async readOne({ userId, trackedPhone }) {
+        const sql = 'SELECT * FROM tracked_phones WHERE user_id = $1 AND tracked_phone = $2';
+        const params = [userId, trackedPhone];
+        const result = await this.db.queryAsync(sql, params).catch((err) => {
+            throw new RepoError(err);
+        });
+        return result && result.rows[0];
+    }
+
+    /**
+     * @method
+     * @param {number} userId - user identifier
+     * @param {boolean} tracked - flag that indicates is the contact tracked
+     * @param {string} trackedPhone - contact added by user that should be tracked
+     * @param {string} sessionId - identifier of session that tracks this contact
+     * @param {boolean} withSession - flag that indicates does contact have session
+     * @return {Promise<Object>} returns data saved in DB
+     */
+    async read({ userId, tracked, trackedPhone, sessionId, withSession }) {
+        let sql = 'SELECT * FROM tracked_phones';
+        let params = '';
+
+        if (typeof userId === 'number') {
+            params += ` user_id = ${userId}`;
+        }
+
+        if (typeof tracked === 'boolean') {
+            params !== '' ? (params += ` AND tracked = ${tracked}`) : (params += ` tracked = ${tracked}`);
```

Comment:
params += params ? ' AND' : '';
params += ` tracked = ${tracked}`;

Reply:
done